src/System.Private.ServiceModel/tests/Scenarios/Extensibility/WebSockets/WebSocketTests.cs

```diff
@@ -0,0 +1,234 @@
+        }
+    }
+
+    [Fact]
+    [OuterLoop]
+	[ActiveIssue(424)]
+    public static void WebSocketHttpsDuplexBinaryStreamed()
+    {
+        BinaryMessageEncodingBindingElement binaryMessageEncodingBindingElement = null;
+        HttpsTransportBindingElement httpsTransportBindingElement = null;
+        CustomBinding binding = null;
+        ClientReceiver clientReceiver = null;
+        InstanceContext context = null;
+        DuplexChannelFactory<IWSDuplexService> channelFactory = null;
+        IWSDuplexService client = null;
+
+        try
+        {
+            // *** SETUP *** \\
+            binaryMessageEncodingBindingElement = new BinaryMessageEncodingBindingElement();
+            httpsTransportBindingElement = new HttpsTransportBindingElement();
+            httpsTransportBindingElement.WebSocketSettings.TransportUsage = WebSocketTransportUsage.Always;
+            httpsTransportBindingElement.MaxReceivedMessageSize = ScenarioTestHelpers.DefaultMaxReceivedMessageSize;
+            httpsTransportBindingElement.MaxBufferSize = ScenarioTestHelpers.DefaultMaxReceivedMessageSize;
+            httpsTransportBindingElement.TransferMode = TransferMode.Streamed;
+            binding = new CustomBinding(binaryMessageEncodingBindingElement, httpsTransportBindingElement);
+
+            clientReceiver = new ClientReceiver();
+            context = new InstanceContext(clientReceiver);
+
+            channelFactory = new DuplexChannelFactory<IWSDuplexService>(context, binding, Endpoints.WebSocketHttpsDuplexBinaryStreamed_Address);
+            client = channelFactory.CreateChannel();
+
+            // *** EXECUTE *** \\
+            using (Stream stream = client.DownloadStream())
+            {
+                int readResult;
+                // Read from the stream, 1000 bytes at a time.
+                byte[] buffer = new byte[1000];
+                do
+                {
+                    readResult = stream.Read(buffer, 0, buffer.Length);
+                }
+                while (readResult != 0);
+            }
+
+            FlowControlledStream uploadStream = new FlowControlledStream();
+            uploadStream.ReadThrottle = TimeSpan.FromMilliseconds(500);
+            uploadStream.StreamDuration = TimeSpan.FromSeconds(1);
+
+            client.UploadStream(uploadStream);
+            client.StartPushingStream();
+            // Wait for the callback to get invoked before telling the service to stop streaming.
+            // This ensures we can read from the stream on the callback while the NCL layer at the service
+            // is still writing the bytes from the stream to the wire.  
+            // This will deadlock if the transfer mode is buffered because the callback will wait for the
+            // stream, and the NCL layer will continue to buffer the stream until it reaches the end.
+
+            clientReceiver.ReceiveStreamInvoked.WaitOne();
+            clientReceiver.ReceiveStreamInvoked.Reset();
```

Comment:
I don't we should have any infinite WaitOne() calls in case a test failure.  Recommend using ScenarioTestHelpers.MaxTestTimeout.  And also checking the return value of WaitOne, where a 'false' means it timed out.